src/util/input-types.ts

```diff
@@ -0,0 +1,10 @@
+import {Observable, ObservableInput} from '../Observable';
+
+export type _mapProject<T, R> = (value: T, index: number) => R;
```

Comment:
one thing though, as other signature does I'd like to suggest name these such as `predicateSignature`, etcs to have clarity since it's not actual predicate but signature of predicate. Any suggestions?